src/game/Strategic/Strategic_AI.cc

```diff
@@ -137,8 +138,7 @@ extern UINT8 gubNumGroupsArrivedSimultaneously;
 //group.  When the queen wants to send forces to attack a town that is defended, the initial number of forces that
 //she would send would be considered too weak.  So, instead, she will send that force to the sector's adjacent sector,
 //and stage, while
-UINT8 *gubGarrisonReinforcementsDenied = NULL;
-UINT8 *gubPatrolReinforcementsDenied = NULL;
+std::unique_ptr<UINT8 []> gubGarrisonReinforcementsDenied;
```

Comment:
Couln't we just use a `std::vector` instead?

Reply:
Sure, but we wouldn't use any of vector's features, most notably its flexibility regarding its size. This array always has as many elements as are defined in army-garrison-groups.json. In theory we could even get rid of the allocation in LoadStragicAI and just reuse the already allocated memory but that seems one micro-optimization step too far to me.